### DISCUSSION TITLE
`F,`missing from Floating Point word set?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#136] [Q&A] [answered]
I've noticed the absence of `F,` from the Forth Draft 19-1 section 12 Floating-Point word set.  Is this intentional?

When the float stack is shared with the data stack, then `,` word would be sufficient, but when there is a separate float stack I don't see how one might append the top of the float stack to the data space.   Are developers meant to define their own version of `F,` when required?  Something like:
```
: F,  ( F: f -- ) HERE 1 CELLS ALLOT F! ;
```
Or should this be added to the next Draft?

**Comment by forthy42** [comment, marked as answer]
``F,`` is found in many Forth systems, and they often also have ``SF,`` and ``DF,`` for single and double floats (just like ``SF!`` and ``DF!``)

Open a proposal on https://forth-standard.org, and it has a good chance to get accepted.

**Comment by ruv** [comment]
A portable implementation for `F,`:

```
: F, ( F: r -- )  FALIGN HERE ( f-addr ) F! 1 FLOATS ALLOT ;
```

See [12.3.5 Address alignment](https://forth-standard.org/standard/float#subsection.12.3.5):
- > a program shall align the address (to be float aligned, single-float aligned, or double-float aligned) before accessing floating-point data at the address.
